### PR TITLE
Simplify JS preprocessor so it always takes a filename. NFC

### PR DESCRIPTION
**Pull request by sbc100**
I noticed this nice simplification while working on #18520

### CODE REVIEW

**Comment by sbc100** [comment]
Depends on #18520

File: src/parseTools.js [hunk 1]
```diff
@@ -94,16 +93,15 @@ function preprocess(text, filenameHint) {
               }
             }
             const after = trimmed.substring(trimmed.indexOf(' '));
-            const truthy = !!vm.runInThisContext(after, { filename: filenameHint, lineOffset: i, columnOffset: line.indexOf(after) });
+            const truthy = !!vm.runInThisContext(after, { filename: filename, lineOffset: i, columnOffset: line.indexOf(after) });
```
Comment (RReverser) [inline src/parseTools.js:96]:
```suggestion
            const truthy = !!vm.runInThisContext(after, { filename, lineOffset: i, columnOffset: line.indexOf(after) });
```

File: src/parseTools.js [hunk 2]
```diff
@@ -56,8 +55,8 @@ function preprocess(text, filenameHint) {
   const showStack = [];
   const showCurrentLine = () => showStack.every((x) => x == SHOW);
 
-  currentlyParsedFilename = filenameHint;
-  const fileExt = (filenameHint) ? filenameHint.split('.').pop().toLowerCase() : '';
+  currentlyParsedFilename = filename;
```
Comment (RReverser) [inline src/parseTools.js:58]: Where is `currentlyParsedFilename` used / is it still necessary?

**Review by RReverser — APPROVED** [review]

File: src/parseTools.js (hunk 2 above)
Comment (sbc100) [reply src/parseTools.js:58]: I think this is global which is used elsewhere.. we may want to try to remove it, but I don't think its related to this change.

Comment (RReverser) [reply src/parseTools.js:58]: Yeah it's not, just noticed here.

**Pull request merged**